components/layout_2020/flexbox/layout.rs

```diff
@@ -2039,18 +2035,27 @@ impl FlexItem<'_> {
                     }
                 }
 
+                let cache = self.box_.block_content_size_cache.borrow_mut().take();
```

Comment:
> Make the computed layout Clone. Then the cache can be used more than once.

Currently, we don't use layout more than once. This would also lead to some issues once layouts are fully connected to the DOM tree. What does it mean for a DOM node to have more than one layout and which one is incrementally updated once incremental layout is in place?

> Also cache the "final layout" in the case that there is a cache miss. That can then be used for subsequent relayouts.

This is essentially incremental layout -- which is a huge task, but it cannot be done exactly at this level. Caching here is done on the box tree, but the box tree is destroyed in nonincremental layout so the cache would also be destroyed. We have plans to implement incremental layout though, so you are on the right track to suggest it.